### PR TITLE
amethyst_core: Reimplement the hide hierarchy system in legion and add unit tests to it

### PR DESCRIPTION
**Pull request by grzi**
## Description

Re implementation of the hide hierarchy system in legion and unit tests 
I hope it's not a too naïve approch to this. open for reviews. 

## Additions

- Unit tests to validate the system

## Removals

- Spec implementation

## Modifications

- Rename to hide_hierarchy_system 
- Switch to legion

## PR Checklist

By placing an x in the boxes I certify that I have:

- [X] Added unit tests for new code added in this PR.
- [X] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Updated the content of the book if this PR would make the book outdated.

If this modified or created any rs files:

- [ ] Ran `cargo +stable fmt --all`
- [ ] Ran `cargo clippy --workspace --features "empty"` (may require `cargo clean` before)
- [ ] Ran `cargo build --features "empty"`
- [ ] Ran `cargo test --workspace --features "empty"`

All of this has been done but due to the legion_v2 state it's actually failing, but not because of my devs

### CODE REVIEW

File: amethyst_core/src/lib.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ pub use self::{
     axis::{Axis2, Axis3},
     event::EventReader,
     hidden::{Hidden, HiddenPropagate},
+    hide_hierarchy_system::build,
```
Comment (ezpuzz) [inline amethyst_core/src/lib.rs:16]: I don't think this is the namespacing we want to use, probably can remove `::build`

File: amethyst_core/src/hide_hierarchy_system.rs [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+};
+
+use crate::HiddenPropagate;
+use crate::transform::{Children, Parent};
+use std::collections::HashSet;
+
+/// This system adds a [HiddenPropagate](struct.HiddenPropagate.html)-component to all children
+/// of an entity with a [HiddenPropagate](struct.HiddenPropagate.html) and removes it when it is removed
+/// from the parent.
+pub fn build() -> impl Runnable {
```
Comment (ezpuzz) [inline amethyst_core/src/hide_hierarchy_system.rs:12]: This is awesome! Thanks for adding tests 👍🏻   Way better than my attempt in #2536

**Review by ezpuzz — CHANGES_REQUESTED** [review]
Thanks!

**Comment by AlveLarsson** [comment]
Did you check out the LT repo we have? IDK if it's still valid or not or if this transform PR is better?
https://github.com/amethyst/legion_transform (and the readme is outdated)

**Review by ezpuzz — APPROVED** [review]

**Comment by ezpuzz** [comment]
@AlveLarsson probably would be good to review legion_transform as a possible replacement after we port the existing implementations.  also I don't see anything about managing hidden state in legion_transform.

**Comment by grzi** [comment]
@AlveLarsson The current system doesn't belong to the transform part IMHO, it's an additional feature specific to amethyst.

Yeah, legion_transform could replace the actual transform module as the code is nearly the same. Maybe as @ezpuzz said, after all the port is done we could 'merge' what has been done on amethyst (if there are) on the legion_transform project and use it as a replacement

**Review by AlveLarsson — APPROVED** [review]
Oh right that wat was what we said, first just a port of the normal transform and after 0.16 the transform repo. Thanks for this! Keep up the great work :)

**Pull request merged**